src/app/(app)/course/[courseId]/students/(hooks)/useStudents.tsx

```diff
@@ -138,16 +141,18 @@ const useStudentsProvider = (): StudentsContextType => {
 
   useEffect(() => {
     const fetchSections = async () => {
-      const sectionsResponse = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/v1/course/${FIXED_COURSE_NUM}/sections`,)
+      const sectionsResponse = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/v1/course/${courseId}/sections`,)
       const sectionsData = await sectionsResponse.json()
       const sectionsToDisplay: DropdownOption[] = sectionsData.map((section: any) => ({
         label: section.name,
         value: String(section.id),
       }),)
       setAllSections(sectionsToDisplay)
     }
-    fetchSections()
-  }, [setAllSections])
+    if (courseId && !isNaN(Number(courseId)) && Number(courseId) > 0) {
```

Comment:
yeah like the courseId would already be valid at this point if you use the one that comes from useCourse?